### PR TITLE
Hive integration with Presto

### PR DESCRIPTION
**Pull request by imakhlin**
Adding containers for:
- Hive 2.3.3
- Postgres 11.1.0

### CODE REVIEW

File: services/postgresql/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ARG POSTGRES_GID
+
+ARG PG_HOME
+
+FROM bitnami/postgresql:${POSTGRES_VERSION}
+
+USER root
+
+# Note, base image assumes that "postgres" user have UID/GID=1001
+ENV POSTGRES_UID=${POSTGRES_UID:-1001} \
```
[the remark this answers is not in the document]
Comment (imakhlin) [reply services/postgresql/docker/Dockerfile]: I tried to make it one statement but unfortunately I wasn't able to refer onto the environment variable that has been declared at the same ENV block

[the remark this answers is not in the document]
Comment (bivas) [reply services/postgresql/docker/Dockerfile]: Use single env per line

File: services/hive/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Set hive metastore database type
+ENV HIVE_DB_TYPE=${HIVE_DB_TYPE}
+
+USER root
+
+COPY --chown=1000:1000 --from=hive-base /opt/hive $HIVE_HOME
+COPY --chown=1000:1000 --from=hive-base /usr/local/bin/startup.sh /usr/local/bin/startup.sh
+COPY --chown=1000:1000 --from=hive-base /usr/local/bin/entrypoint.sh /usr/local/bin/entrypoint.sh
+
+ADD "https://jdbc.postgresql.org/download/postgresql-${PSQL_JDBC_VERSION}.jar" ${HIVE_HOME}/lib/
```
Comment (bivas) [inline services/hive/docker/Dockerfile]: Use a single `RUN` statement to create single layer

File: services/postgresql/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ARG POSTGRES_UID=1001
+ARG POSTGRES_GID=1001
+
+# Allow control over the NAMI logging level. Set to "trace" for debug.
+ARG NAMI_LOG_LEVEL=info
+
+ENV NAMI_LOG_LEVEL=${NAMI_LOG_LEVEL}
+
+# Note, PGDATA should refer to the mount point.
+ENV PGDATA=/bitnami/postgresql
```
Comment (bivas) [inline services/postgresql/docker/Dockerfile]: Should be `/postgresql/data`

File: services/hive/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+COPY --chown=1000:1000 --from=hive-base /usr/local/bin/entrypoint.sh /usr/local/bin/entrypoint.sh
+
+RUN set -e && \
+    wget --quiet "https://jdbc.postgresql.org/download/postgresql-${PSQL_JDBC_VERSION}.jar" \
+      -P ${HIVE_HOME}/lib/ && \
+    wget --quiet "https://downloads.mariadb.com/Connectors/java/connector-java-${MYSQL_JDBC_VERSION}/mariadb-java-client-${MYSQL_JDBC_VERSION}.jar" \
+      -P ${HIVE_HOME}/lib/ &&\
+    chown -R ${CONTAINER_USER}:${CONTAINER_USER} ${HIVE_HOME}
+
+USER ${CONTAINER_USER}
```
Comment (bivas) [inline services/hive/docker/Dockerfile]: You can switch to that user *before* your `RUN` section and avoid the `chown` command

File: services/postgresql/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    chown -R $POSTGRES_UID:$POSTGRES_GID /bitnami/postgresql && \
+    chown -R $POSTGRES_UID:$POSTGRES_GID /.nami && \
+    mkdir $PGDATA/data && \
+    chown -R $POSTGRES_UID:$POSTGRES_GID $PGDATA
+
+USER $POSTGRES_UID
+
+ENV HOME /home/$POSTGRES_UNAME
+
+RUN set -x && \
```
Comment (bivas) [inline services/postgresql/docker/Dockerfile]: These aliases are already in base images - please remove this layer

File: services/postgresql/docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+RUN echo "Running as user: $(id -u):$(id -g)"
+
+USER root
+
+# Create iguazio user/group
+RUN set -ex && \
+    groupadd -g $POSTGRES_UID $POSTGRES_UNAME && \
+    useradd --uid $POSTGRES_UID --gid $POSTGRES_UNAME --home /home/$POSTGRES_UNAME --shell /bin/bash --create-home $POSTGRES_UNAME
+
+RUN chown -R $POSTGRES_UID:$POSTGRES_GID /home/$POSTGRES_UNAME && \
```
Comment (bivas) [inline services/postgresql/docker/Dockerfile]: don't create another layer, fuse them together to single layer.

Comment (bivas) [inline services/postgresql/docker/Dockerfile]: No need to `chown` here - you already created it with this user.

File: services/postgresql/docker/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    chown -R $POSTGRES_UID:$POSTGRES_GID /opt/bitnami && \
+    chown -R $POSTGRES_UID:$POSTGRES_GID /bitnami/postgresql && \
```
Comment (bivas) [inline services/postgresql/docker/Dockerfile]: This is your `$PGDATA` folder...

File: services/postgresql/docker/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+USER root
+
+# Create iguazio user/group
+RUN set -ex && \
+    groupadd -g $POSTGRES_UID $POSTGRES_UNAME && \
+    useradd --uid $POSTGRES_UID --gid $POSTGRES_UNAME --home /home/$POSTGRES_UNAME --shell /bin/bash --create-home $POSTGRES_UNAME
+
+RUN chown -R $POSTGRES_UID:$POSTGRES_GID /home/$POSTGRES_UNAME && \
+    chown -R $POSTGRES_UID:$POSTGRES_GID /opt/bitnami && \
```
Comment (bivas) [inline services/postgresql/docker/Dockerfile]: can this be at `/postgresql`?

File: services/postgresql/docker/Dockerfile (hunk 5 above)
Comment (imakhlin) [reply services/postgresql/docker/Dockerfile]: Note that this image is not based on our base images but on `bitnami`

File: services/postgresql/docker/Dockerfile (hunk 8 above)
Comment (imakhlin) [reply services/postgresql/docker/Dockerfile]: no, unless I change their `run.sh`
see https://github.com/bitnami/bitnami-docker-postgresql/blob/master/11/debian-9/rootfs/run.sh

File: services/postgresql/docker/Dockerfile (hunk 5 above)
Comment (bivas) [reply services/postgresql/docker/Dockerfile]: We only use these images as binary cache. The real images are elsewhere.

**Comment by imakhlin** [comment]
Replaced with PR #25

**Pull request closed without merging**